task-manager/src/components/ErrorBoundary.js

```diff
@@ -0,0 +1,27 @@
+import React, { Component } from 'react';
+import './ErrorBoundary.css';
+
+class ErrorBoundary extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { hasError: false };
+  }
+
+  static getDerivedStateFromError(error) {
```

Comment:
The getDerivedStateFromError method is static because React calls this method at the class level to handle errors and update state uniformly across all instances of the component. By Using a static method ensures  us that the error handling behavior is consistent and does not rely on instance-specific details. It is static because it doesn’t operate on individual component instances but on the class as a whole. It is used to update the state of the component when an error is thrown in one of its descendants.